ark/phenotyping/som_utils.py

```diff
@@ -19,6 +20,95 @@
 from ark.utils import misc_utils
 
 
+def calculate_channel_percentiles(tiff_dir, fovs, channels, img_sub_folder, percentile):
+    """Calculates average percentile for each channel in the dataset
+
+    Args:
+        tiff_dir (str):
+            Name of the directory containing the tiff files
+        fovs (list):
+            List of fovs to include
+        channels (list):
+            List of channels to include
+        img_sub_folder (str):
+            Sub folder within each FOV containing image data
+        percentile (float):
+            The specific percentile to compute
+
+    Returns:
+        dict:
+            The mapping between each channel and its normalization value
+    """
+
+    # create dict to hold all percentiles
+    percentile_dict = {}
+
+    # loop over channels and FOVs
+    for channel in channels:
+        percentile_list = []
+        for fov in fovs:
+
+            # load image data and remove 0 valued pixels
+            img = load_utils.load_imgs_from_tree(data_dir=tiff_dir, img_sub_folder=img_sub_folder,
+                                                 channels=[channel], fovs=[fov]).values[0, :, :, 0]
+            img = img[img > 0]
+
+            # record and store percentile
+            # NOTE: if there are no non-zero pixels, assign directly to 0 to avoid empty list error
+            img_percentile = np.quantile(img, percentile) if len(img) > 0 else 0
```

Comment:
I think it makes more sense to completely skip a blank channel, rather than counting it as a zero for the purposes of constructing the average